### PR TITLE
fix: update generated workflows to use latest versions

### PR DESCRIPTION
**Pull request by wolfy1339**

### CODE REVIEW

File: lib/create-test-action.js [hunk 1]
```diff
@@ -18,20 +18,15 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node_version: ["12", "14", "16"]
+        node_version: ["14", "16", "18"]
```
Comment (kfcampbell) [inline lib/create-test-action.js:21]: Has there been a statement about what node versions we're supporting? Are 14, 16, and 18 generally considered by the community to be the supported versions? (Sorry, I've been out of the Node ecosystem a little bit.)

File: lib/create-release-action.js [hunk 2]
```diff
@@ -21,16 +21,11 @@ jobs:
...
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
         with:
           node-version: "lts/*"
-      - uses: actions/cache@v1
-        with:
-          path: ~/.npm
-          key: \${{ runner.os }}-node-\${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            \${{ runner.os }}-node-
```
Comment (kfcampbell) [inline lib/create-release-action.js:33]: Do you mind explaining a little more about the reason for the above caching changes? Presumably it's due to the 2 major version upgrade, though I couldn't find an obvious culprit in the recommended [README](https://github.com/actions/cache/blob/main/examples.md#node---npm) [diffs](https://github.com/actions/cache/blob/v1/examples.md#node---npm).

File: lib/create-test-action.js (hunk 1 above)
Comment (wolfy1339) [reply lib/create-test-action.js:21]: This is inline with all the other current JS octokit repositories.

v18 is the latest LTS release
v16 the prior LTS
v14 is the LTS before then, the latest LTS release to support ESM modules

From the history of the JS octokit modules, it seems it's always been the current LTS, and the 2 prior ones

We had a discussion about dropping NodeJS v10 in https://github.com/octokit/oauth-app.js/issues/299 and further in the discussion also decided on dropping v12

File: lib/create-release-action.js (hunk 2 above)
Comment (wolfy1339) [reply lib/create-release-action.js:33]: This is the source of the changes
https://github.com/octokit/openapi/commit/0420264dc65e564d72e5efba4031a5ab1c69ccad

`actions/cache` was dropped in favour of using the cache in `actions/setup-node`

**Review by kfcampbell — APPROVED** [review]
Thank you for answering my questions!

**Pull request merged**